merkle-tree/src/lib.rs

```diff
@@ -1 +1,60 @@
+use serde::{Deserialize, Serialize};
+use simperby_common::crypto::Hash256;
+use simperby_common::MerkleProof;
+use simperby_kv_storage::KVStorage;
+use std::marker::PhantomData;
+use thiserror::Error;
 
+#[derive(Error, Debug, Serialize, Deserialize, Clone)]
+pub enum Error {
+    /// When the given storage is not a valid Merkle-tree-indexed database.
+    #[error("invalid storage {0}")]
+    MalformedStorage(String),
+    /// When the provided storage emitted an error during the operation.
+    #[error("storage error: {0}")]
+    StorageError(simperby_kv_storage::Error),
+}
+
+/// A state machine that interpret the given storage as a Merkle tree rooted by a specific key.
+///
+/// Note that there is no `get()` method as leaf nodes (which the user of this struct would like to read)
+/// are stored by their keys directly, so one can just access directly to the underlying `KVStore` using the key.
+pub struct MerkleTree<S> {
+    // TODO: add the fields (for real) and remove _x (dummy)
+    _x: PhantomData<S>,
+}
+
+impl<S: KVStorage> MerkleTree<S> {
```

Comment:
Maybe we need a sort of `contain(value) -> bool` method?

Reply:
fixed.

Reply:
no, for the same reason with `get()`.